core/components/molecules/emptyState/EmptyState.tsx

```diff
@@ -3,94 +3,191 @@ import classNames from 'classnames';
+   * Sets the width of the `EmptyState` component.
    */
-  children?: React.ReactNode;
+  width?: number | string;
   /**
-   * Image to be shown. Image height should be set while providing image. Provide height as 128px when size is  'small'. Provide height as 256px when size is  'large'.
+   * child component to be added inside `EmptyState`
    */
-  image?: React.ReactNode;
+  children?: React.ReactNode;
 }
 
 export const imageHeight = {
   large: '256px',
   small: '128px',
+  standard: '256px',
+  compressed: '256px',
+  tight: '256px',
 };
 
-export const HeadingSize: Record<EmptyStateSize, HeadingProps['size']> = {
+export const templateWidth = {
+  standard: '480px',
+  compressed: '400px',
+  tight: '320px',
+  large: '480px',
+  small: '480px',
+};
+
+export const HeadingSize: Record<TEmptyStateSize, HeadingProps['size']> = {
   large: 'l',
   small: 'm',
+  standard: 'l',
+  compressed: 'l',
+  tight: 'l',
 };
 
-export const textSize: Record<EmptyStateSize, TextProps['size']> = {
+export const textSize: Record<TEmptyStateSize, TextProps['size']> = {
   large: 'large',
   small: 'regular',
+  standard: 'large',
+  compressed: 'large',
+  tight: 'regular',
 };
 
 export const EmptyState = (props: EmptyStateProps) => {
-  const { imageSrc, title, description, size, children, className, image } = props;
+  const {
+    imageSrc,
+    title,
+    description,
+    size = 'standard',
+    children,
+    className,
+    image,
+    maxWidth,
+    minWidth,
+    width,
+  } = props;
 
   const baseProps = extractBaseProps(props);
 
-  const WrapperClass = classNames(
+  let templateSize: TEmptyStateSize = 'standard';
+
+  const isValidSize = (size: TEmptyStateSize) => size === 'large' || size === 'small';
+
+  if (title || description) {
+    templateSize = isValidSize(size) ? size : 'large';
+  } else {
+    templateSize = isValidSize(size) ? 'standard' : size;
+  }
+
+  const wrapperClasses = classNames(
     {
       ['EmptyState']: true,
     },
     className
   );
 
-  const HeadingClass = classNames({
+  const emptyStateWrapper = classNames({ ['EmptyState-Wrapper']: true }, className);
+
+  const headingClasses = classNames({
     ['EmptyState-title']: true,
-    [`EmptyState-title--${size}`]: true,
+    [`EmptyState-title--${templateSize}`]: true,
   });
 
-  const TextClass = classNames({
+  const textClasses = classNames({
     ['EmptyState-description']: true,
-    [`EmptyState-description--${size}`]: children !== undefined,
+    [`EmptyState-description--${templateSize}`]: children !== undefined,
   });
 
+  if (title || description) {
+    return (
+      <div data-test="DesignSystem-EmptyState" {...baseProps} className={wrapperClasses}>
+        {image && <div style={{ height: imageHeight[templateSize] }}>{image}</div>}
+        {imageSrc && !image && (
+          //TODO(a11y)
+          //eslint-disable-next-line
+          <img src={imageSrc} height={imageHeight[templateSize]} data-test="DesignSystem-EmptyState--Img" />
```

Comment:
Instead of avoiding a11y warning, please add an `alt` prop to the image.

Reply:
done

Reply:
I'm not able to see requested changes here. Please check it again

Reply:
this is the deprecated EmptyState do you want me to change it here also?
as we have deprecated all old props